File: frontend/src/types/domain.ts
/**
 * PRD 0514 도메인 타입 (FE).
 *
 * 새 메인 대시보드 통합 응답 + 추천 질문 Agent v1.1 구조.
 */

export type DateString = string;
export type DateTimeString = string;
export type UrlString = string;
export type Direction = 'UP' | 'DOWN';

// ── 고객사 프로필 (PRD 0514) ─────────────────────────────
export interface CustomerProfile {
  customer_id: string;
  industry: string;
  market_region: string;
  product_group: string[];
  sensitive_topics: string[];
  risk_factors: string[];
}

// ── Top Mover (차트1) ────────────────────────────────────
export interface IndicatorPoint {
  date: string;
  value: number;
}

export interface TopMover {
  indicator: string;
  value: number;
  unit: string;
  change_d1: number | null;
  change_w1: number;
  change_m1: number | null;
  score: number;
  series: IndicatorPoint[];
}

// ── Cause Flow (차트2) ───────────────────────────────────
export interface FlowEvidence {
  news_id: string;
  title: string;
  date: string;
  url: string;
}

export interface CauseFlowStep {
  step: number;
  node: string;
  evidence: FlowEvidence[];
}

// ── Interpretation (What/Why/Impact) ─────────────────────
export interface ImpactItem {
  risk_factor: string;
  direction: '증폭' | '완화' | '중립';
  reason: string;
}

export interface Interpretation {
  what: string;
  why: string;
  impact: ImpactItem[];
}

// ── Strategy ─────────────────────────────────────────────
export interface Strategy {
  strategy_summary: string;
  recommended_actions: string[];
  negotiation_points: string[];
}

// ── 통합 Dashboard ───────────────────────────────────────
export interface DashboardPayload {
  customer: string;
  product: string;
  generated_at: DateTimeString;
  chart1_top_movers: TopMover[];
  chart2_cause_flow: CauseFlowStep[];
  interpretation: Interpretation;
  strategy: Strategy;
}

// ── 추천 질문 (SMI-Bot v1.1) ─────────────────────────────
export interface TodayQuestion {
  qid: string;
  text: string;
  trigger_indicators: string[];
  related_groups_internal: string[];
  score: number;
}

export interface TodayQuestionsPayload {
  product: string;
  generated_at: DateTimeString;
  questions: TodayQuestion[];
}

// ── 답변 (SMI-Bot v1.1) ──────────────────────────────────
export interface QuestionAnswerSources {
  indicators: string[];
}

export interface QuestionAnswer {
  qid: string;
  briefing: string;
  sales_rep_script: string;
  sources: QuestionAnswerSources;
  confidence: number;
}

// ── 세션 ─────────────────────────────────────────────────
export type UserRole = 'sales' | 'manager' | 'admin';

export interface SessionUser {
  user_id: string;
  user_role: UserRole;
  name?: string;
}

File: backend/app/schemas/domain.py
"""PRD 0514 Pydantic 도메인 타입.

shared/types/*.ts 일부는 더 이상 사용하지 않음 (negotiation_style 제거).
새 PRD 출력 스키마(MCS-Advisor + SMI-Bot v1.1) 기준.
"""
from __future__ import annotations

from datetime import date as DateT
from datetime import datetime
from enum import StrEnum
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class Direction(StrEnum):
    UP = "UP"
    DOWN = "DOWN"


class UserRole(StrEnum):
    SALES = "sales"
    MANAGER = "manager"
    ADMIN = "admin"


class AssignmentRole(StrEnum):
    PRIMARY = "primary"
    SUPPORT = "support"


class _ORMModel(BaseModel):
    model_config = ConfigDict(from_attributes=True)


class CustomerProfile(_ORMModel):
    customer_id: str
    industry: str
    market_region: str
    product_group: list[str]
    sensitive_topics: list[str]
    risk_factors: list[str]


class AxisGroup(BaseModel):
    indicators: list[str]
    weights: list[float]


class Product(_ORMModel):
    code: str
    key_features: list[str]
    key_feature_importance: list[float]
    axis: dict[str, AxisGroup]


# ── Top Mover (메인 대시보드 차트1) ────────────────────────
class TopMover(BaseModel):
    indicator: str
    value: float
    unit: str
    change_d1: float | None = None
    change_w1: float
    change_m1: float | None = None
    score: float
    series: list[IndicatorPointOut] = Field(default_factory=list)


# ── Cause Flow (메인 대시보드 차트2) ───────────────────────
class FlowEvidence(BaseModel):
    news_id: str
    title: str
    date: str
    url: str


class CauseFlowStep(BaseModel):
    step: int
    node: str
    evidence: list[FlowEvidence] = Field(default_factory=list)


# ── LLM 해석 (메인 대시보드 What/Why/Impact) ────────────────
class ImpactItem(BaseModel):
    risk_factor: str
    direction: Literal["증폭", "완화", "중립"]
    reason: str


class Interpretation(BaseModel):
    what: str
    why: str
    impact: list[ImpactItem]


# ── 전략 (메인 대시보드 전략/추천행동/협상포인트) ────────────
class Strategy(BaseModel):
    strategy_summary: str
    recommended_actions: list[str] = Field(min_length=1, max_length=5)
    negotiation_points: list[str] = Field(min_length=1, max_length=5)


# ── 추천 질문 (SMI-Bot v1.1) ────────────────────────────────
class TodayQuestion(BaseModel):
    qid: str
    text: str
    trigger_indicators: list[str]
    related_groups_internal: list[str]
    score: float


# ── 답변 (SMI-Bot v1.1) ─────────────────────────────────────
class AnswerSources(BaseModel):
    indicators: list[str]


class QuestionAnswer(BaseModel):
    qid: str
    briefing: str
    sales_rep_script: str
    sources: AnswerSources
    confidence: float


# ── 뉴스 (인과 흐름도 evidence) ──────────────────────────────
class NewsDoc(BaseModel):
    source: str
    title: str
    summary: str
    url: str
    published_at: datetime


# ── 지표 시계열 ──────────────────────────────────────────────
class IndicatorPointOut(BaseModel):
    date: DateT
    value: float


class IndicatorSeries(BaseModel):
    feature_name: str
    unit: str
    points: list[IndicatorPointOut]


# ── 세션 ────────────────────────────────────────────────────
class SessionUser(BaseModel):
    user_id: str
    user_role: UserRole
    name: str | None = None
